nixos/modules/services/networking/prosody.nix

```diff
@@ -61,12 +62,38 @@ let
       description = "Service discovery";
     };
 
-    legacyauth = mkOption {
+    # Not essential, but recommended
+    carbons = mkOption {
       type = types.bool;
       default = true;
-      description = "Legacy authentication. Only used by some old clients and bots";
+      description = "Keep multiple clients in sync";
+    };
+
+    pep = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Enables users to publish their mood, activity, playing music and more";
+    };
+
+    private = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Private XML storage (for room bookmarks, etc.)";
+    };
+
+    blocklist = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Allow users to block communications with other users";
     };
 
+    vcard = mkOption {
```

Comment:
There is a lot of new options, but they are not referenced anywhere in module, so they will not work. What's the purpose of that change then?

Reply:
Every option in `moduleOpts` is automatically converted  to lua and added to the config file at https://github.com/NixOS/nixpkgs/pull/32960/files#diff-d51fea555a73d3c752f162a7a433c498R442 so they don't need to be explicitly mentioned anywhere else in the module, avoiding loads of boilerplate code. Note that this mechanism was already in place before this PR. :wink:

Reply:
@florianjacob Very interesting, thank you!